plugins/base/src/main/kotlin/transformers/documentables/utils/ClassGraphBuilder.kt

```diff
@@ -0,0 +1,90 @@
+package org.jetbrains.dokka.base.transformers.documentables.utils
+
+import com.intellij.psi.PsiClass
+import kotlinx.coroutines.*
+import org.jetbrains.dokka.DokkaConfiguration
+import org.jetbrains.dokka.analysis.DescriptorDocumentableSource
+import org.jetbrains.dokka.analysis.PsiDocumentableSource
+import org.jetbrains.dokka.analysis.from
+import org.jetbrains.dokka.links.DRI
+import org.jetbrains.dokka.model.*
+import org.jetbrains.dokka.utilities.parallelForEach
+import org.jetbrains.kotlin.descriptors.ClassDescriptor
+import org.jetbrains.kotlin.types.KotlinType
+import org.jetbrains.kotlin.types.typeUtil.immediateSupertypes
+import org.jetbrains.kotlin.types.typeUtil.isAnyOrNullableAny
+import java.util.concurrent.ConcurrentHashMap
+
+
+/**
+ * The class allows lo build a full class hierarchy via descriptors
+ */
+class ClassGraphBuilder {
+    suspend operator fun invoke(original: DModule): SourceSetDependent<Map<DRI, List<DRI>>> = coroutineScope {
+        val map = original.sourceSets.associateWith { ConcurrentHashMap<DRI, List<DRI>>() }
+        original.packages.parallelForEach { visitDocumentable(it, map) }
+        map
+    }
+
+    private suspend fun collectSupertypesFromKotlinType(
+        DRIWithKType: Pair<DRI, KotlinType>,
```

Comment:
it's not a class name, so maybe start with lowercase? `driWithType`. This and below

I thought it was an object while reading the code

Also, I think it would not hurt readability to deconstruct it instead of using `.first` and `.second` everywhere
val (dri, kotlinType) = driWithKType